src/main/kotlin/no/nb/mlt/wls/application/hostapi/order/ApiCreateOrderPayload.kt

```diff
@@ -0,0 +1,137 @@
+        accessMode = READ_ONLY
+    )
+    val orderLine: List<OrderLine>,
+    @Schema(
+        description = """Describes what type of order this is.
+            "LOAN" means that the order is for borrowing items to external or internal users,
+            usually meaning the items will be viewed, inspected, etc.
+            "DIGITIZATION" means that the order is specifically for digitizing items,
+            usually meaning that the order will be delivered to digitization workstation.""",
+        examples = ["LOAN", "DIGITIZATION"]
+    )
+    val orderType: Order.Type,
+    @Schema(
+        description = """Who to contact in relation to the order if case of any problems/issues/questions.""",
+        example = "Dr. Heinz Doofenshmirtz"
+    )
+    val contactPerson: String,
+    @Schema(
+        description = """Address for the order, used in cases where storage operator sends out the order directly.""",
+        example = "{...}"
+    )
+    val address: Order.Address?,
+    @Schema(
+        description = """Notes regarding the order, such as delivery instructions, special requests, etc.""",
+        example = "I need this order in four weeks, not right now."
+    )
+    val note: String?,
+    @Schema(
+        description = """Callback URL to use for sending order updates to the host system.
+            For example when order items get picked or the order is cancelled.""",
+        example = "https://callback-wls.no/order"
+    )
+    val callbackUrl: String
+) {
+    fun toCreateOrderDTO() =
+        CreateOrderDTO(
+            hostName = hostName,
+            hostOrderId = hostOrderId,
+            orderLine = orderLine.map { it.toCreateOrderItem() },
+            orderType = orderType,
+            address = address,
+            contactPerson = contactPerson,
+            note = note,
+            callbackUrl = callbackUrl
+        )
+
+    @Throws(ValidationException::class)
+    fun validate() {
+        if (hostOrderId.isBlank()) {
+            throw ValidationException("The order's hostOrderId is required, and can not be blank")
+        }
+
+        if (orderLine.isEmpty()) {
+            throw ValidationException("The order must have at least one order line")
+        }
+
+        if (!isValidUrl(callbackUrl)) {
+            throw ValidationException("The order's callback URL is required, and must be a valid URL")
+        }
+
+        orderLine.forEach(OrderLine::validate)
+        address?.validate()
+    }
+
+    private fun isValidUrl(url: String): Boolean {
+        // Yes I am aware that this function is duplicated in three places
+        // But I prefer readability over DRY in cases like this
+
+        val validator = UrlValidator(arrayOf("http", "https")) // Allow only HTTP/HTTPS
+        return validator.isValid(url)
+    }
+}
+
+fun Order.toCreateApiOrderPayload() =
+    ApiOrderPayload(
+        hostName = hostName,
+        hostOrderId = hostOrderId,
+        status = status,
```

Comment:
Should be removed (i mean compiler won't accept it if we change to proper class above.